product-catalog/test/productcatalog/pages/ProductDetailPageContentTest.java

```diff
@@ -0,0 +1,134 @@
+package productcatalog.pages;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.neovisionaries.i18n.CountryCode;
+import common.categories.CategoryUtils;
+import common.cms.CmsPage;
+import common.pages.*;
+import common.prices.PriceFinder;
+import common.utils.PriceFormatter;
+import common.utils.Translator;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryTree;
+import io.sphere.sdk.models.Image;
+import io.sphere.sdk.models.ImageDimensions;
+import io.sphere.sdk.products.ProductProjection;
+import io.sphere.sdk.products.ProductVariant;
+import io.sphere.sdk.shippingmethods.ShippingRate;
+import org.javamoney.moneta.Money;
+import org.junit.Test;
+import productcatalog.models.RichShippingRate;
+
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+import static common.JsonUtils.readJsonNodeFromResource;
+import static common.products.ProductUtils.getProductById;
+import static common.products.ProductUtils.getQueryResult;
+import static io.sphere.sdk.json.JsonUtils.readObjectFromResource;
+import static io.sphere.sdk.json.JsonUtils.toJsonNode;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProductDetailPageContentTest {
```

Comment:
I feel in here you are actually testing the factories and not so much this `ProductDetailPageContent`. If that's the case, the tests should go into different test classes.